_config/email.yml

```diff
@@ -1,12 +1,55 @@
 ---
-Name: emailconfig
+Name: mailer
 ---
 SilverStripe\Core\Injector\Injector:
-  Swift_Transport: Swift_MailTransport
-  Swift_Mailer:
-    constructor:
-      - '%$Swift_Transport'
-  SilverStripe\Control\Email\Mailer:
-    class: SilverStripe\Control\Email\SwiftMailer
-    properties:
-      SwiftMailer: '%$Swift_Mailer'
+  Symfony\Component\Mailer\MailerInterface:
+    class: Symfony\Component\Mailer\Mailer
+    constructor:
+      0: '%$Symfony\Component\Mailer\Transport\TransportInterface'
+      1: '%$Symfony\Component\Messenger\MessageBusInterface.mailer'
+  Symfony\Component\Messenger\MessageBusInterface.mailer:
+    class: Symfony\Component\Messenger\MessageBus
+    constructor:
+      0:
+        - '%$Symfony\Component\Messenger\Middleware\MiddlewareInterface.mailer'
+  Symfony\Component\Messenger\Middleware\MiddlewareInterface.mailer:
+    class: Symfony\Component\Messenger\Middleware\SendMessageMiddleware
+    constructor:
+      0: '%$Symfony\Component\Messenger\Transport\Sender\SendersLocatorInterface.mailer'
+      1: '%$Symfony\Component\EventDispatcher\EventDispatcherInterface.mailer'
+  Symfony\Component\Messenger\Transport\Sender\SendersLocatorInterface.mailer:
+    class: Symfony\Component\Messenger\Transport\Sender\SendersLocator
+    constructor:
+      0:
+        '*':
+          - 'Symfony\Component\Messenger\Transport\Sender\SenderInterface.mailer'
+      1: '%$Symfony\Component\DependencyInjection\ContainerInterface.mailer'
+  Symfony\Component\Messenger\Transport\Sender\SenderInterface.mailer:
+    class: SilverStripe\Control\Email\EmailSender
+  Symfony\Component\DependencyInjection\ContainerInterface.mailer:
+    class: Symfony\Component\DependencyInjection\Container
+    calls:
+      - ['set', ['Symfony\Component\Messenger\Transport\Sender\SenderInterface.mailer', '%$Symfony\Component\Messenger\Transport\Sender\SenderInterface.mailer']]
+  Symfony\Component\EventDispatcher\EventDispatcherInterface.mailer:
+    class: Symfony\Component\EventDispatcher\EventDispatcher
+---
+Name: mailer-dsn-default-config
+---
+SilverStripe\Core\Injector\Injector:
+  Symfony\Component\Mailer\Transport\TransportInterface:
+    factory: Symfony\Component\Mailer\Transport
+    factory_method: fromDsn
+    constructor:
+      dsn: 'sendmail://default'
+---
+Name: mailer-dsn-env
+After:
+  - 'mailer-dsn-default-config'
+  - 'mailer-dsn-project-config'
```

Comment:
Have updated.

There is a risk that the MAILER_DSN config won't work if projects also use `After: '*'` for their project email config, however that's kind of on them for not following the instructions which say use 'After: 'mailer-dsn-default-config'`

Also would imply they're attempting to use both custom .yml AND the MAILER_DSN .env variable at the same time, which you shouldn't do

Reply:
> They can use whatever naming convention they want for their own file

Then what is `mailer-dsn-project-config` doing in this `After:`? That's not something we're adding in this PR, it's what the docs suggest as the name for the project config.

Either it isn't needed in this `After:` (in which case we should remove it from here) or it _is_ needed here which means we need to make it _really_ clear in the docs that this is an important name for their project-specific mailer config.

> Cannot update this to After: '*' - it creates a circular dependency when I flush

I can't replicate that.... I've changed it to `After: '*'` locally with a fresh copy of the PR and it works perfectly. This makes sense - there _is_ nothing saying `After: 'mailer-dsn-env'` because it's intended as a global override. So there is no circular dependency.